### DISCUSSION TITLE
Gradual upgrade to 3.0

### DISCUSSION THREAD

**Discussion by AdamKatzDev** [api-platform/api-platform#2359] [Q&A]
Is there a way to gradually migrate to 3.0, i.e. keep old metadata for pre-3.0 resources and new metadata for migrated resources at the same time?
As I understand currently it is required to migrate every resource to the new system at once which is a bit complicated. I'd prefer to roll out such changes gradually.

**Comment by alanpoulain** [comment]
Sorry, it's not possible at the current time. Maybe @soyuka can explain why.

**Reply by soyuka** [reply]
As far a I remember in 2.7 new metadata is working no?

**Reply by AdamKatzDev** [reply]
But we have to migrate every resource simultaneously, it is not that different to migrating straight to 3.0.
For context in my project Output Transformers are used widely, so migrating and testing every resource is a lot of work. I'd like to migrate them in batches of a manageable size.
